fixtures/tests/test_fixture.py

```diff
@@ -162,7 +180,7 @@ def test_details_from_child_fixtures_are_returned(self):
             self.assertEqual({}, parent.getDetails())
             # After cleanup the child details are still gone.
             child.addDetail('foo', 'content')
-        self.assertEqual({}, parent.getDetails())
+        self.assertRaises(TypeError, parent.getDetails)
```

Comment:
It's not clear to me where the `TypeError` comes from. I'm guessing it's the fixture trying to do something to some element of its internal state after that state has been set to `None`. 

On the one hand, I think this is not the most helpful interface. On the other, resetting to `None` is probably the simplest and cleanest implementation. Thoughts?